### PR TITLE
Cleanup CircleCI

### PR DESCRIPTION
**Pull request by chrisballinger**
* Uses CocoaPods 1.7.0rc2 for Swift 4.2 + 5.0 podspec support
* CocoaPods 1.7 Specs CDN
* Use CircleCI 2.1 executors and commands

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -182,43 +87,56 @@ jobs:
...
+            - vendor/bundle
+
+  lint-pod:
+    description: "Lints podspec with specified Swift version"
+    parameters:
+      swift-version:
+        type: string
+        default: "5.0"
+    steps:
+      - run: bundle exec pod repo add-cdn jsDelivr-Specs "https://cdn.jsdelivr.net/cocoa/"
```
Comment (ZevEisenberg) [inline .circleci/config.yml]: #? why this repo, in particular?

Comment (chrisballinger) [reply .circleci/config.yml]: It's the only Specs CDN so far: http://blog.cocoapods.org/CocoaPods-1.7.0-beta/

Comment (ZevEisenberg) [reply .circleci/config.yml]: Ohh, got it. I thought it was going to be a system where any company could sign up to host a mirror, but it’s actually just one free CDN, which is sponsored by a bunch of companies. Smart. https://www.jsdelivr.com

Comment (chrisballinger) [reply .circleci/config.yml]: My guess is that once it's no longer experimental there might be some other CDN mirrors pop up. CDN Specs still doesn't support some use cases like `trunk push`, as far as I'm aware.

**Pull request merged**